### PR TITLE
Remember sidebar items state #4128

### PR DESCRIPTION
**Pull request by oleitao**
**Resolved / Related Issues**
Itemize resolved / related issues by this PR.
- Remember sidebar sections state

**Details of Changes**
Add details of changes here.
- Added into settings preferences an option to allow to enable remember last sections sidebar states

**Validation**
How did you test these changes?
- [x] Change section sidebar states
- [x] Close Files
- [x] Open Files and check if sections states are the same as previous

### CODE REVIEW

**Comment by yaichenbaum** [comment]
> Added into settings preferences an option to allow to enable remember last sections sidebar states

@oleitao This shouldn't be a setting, this should be the default behavior.

**Comment by oleitao** [comment]
> > Added into settings preferences an option to allow to enable remember last sections sidebar states
> 
> @oleitao This shouldn't be a setting, this should be the default behavior.

@yair100 i change remember last sections sidebar states to work as default behaviour.

**Comment by winston-de** [comment]
It only seems to remember the state for the drives and cloud drive sections.

https://user-images.githubusercontent.com/59544401/114632347-64212d00-9c73-11eb-9d2d-d4cb86dc28bb.mp4

**Comment by winston-de** [comment]
Also, is it supposed to remember the state after collapsing the sidebar?

https://user-images.githubusercontent.com/59544401/114632496-a9ddf580-9c73-11eb-9c30-2f444e6019cb.mp4

**Comment by oleitao** [comment]
> Also, is it supposed to remember the state after collapsing the sidebar?
> 
>  2021-04-13-161606_Trim.mp4

@winston-de that's strange...
I did the follow changes on section then closed Files and open again and checked the section was open or closed based on my last action:

- Favorites (open)
- Library (close)
- Drives (open)
- CloudDrives (close)
- Network (open)

----------------------

- Favorites (close)
- Library (open)
- Drives (close)
- CloudDrives (open)
- Network (close)

----------------------

- Favorites (close)
- Library (close)
- Drives (close)
- CloudDrives (close)
- Network (close)

----------------------

- Favorites (open)
- Library (open)
- Drives (open)
- CloudDrives (open)
- Network (open)

----------------------

changes before close Files
Capture1

open again Files
Capture2

**Comment by oleitao** [comment]
> Also, is it supposed to remember the state after collapsing the sidebar?
> 
>  2021-04-13-161606_Trim.mp4

That's strange, as you can see that works fine to me:

test1

test2

**Comment by winston-de** [comment]
Interesting, the first issue seems to be solved after resetting the dev app. However, it still doesn't remember the state after collapsing the sidebar. Do you have any idea why that may be?

File: Files/ViewModels/SettingsViewModels/PreferencesViewModel.cs [hunk 1]
```diff
@@ -17,8 +17,8 @@ public class PreferencesViewModel : ObservableObject
         private Terminal selectedTerminal = App.AppSettings.TerminalController.Model.GetDefaultTerminal();
         private bool pinRecycleBinToSideBar = App.AppSettings.PinRecycleBinToSideBar;
         private bool showConfirmDeleteDialog = App.AppSettings.ShowConfirmDeleteDialog;
-        private bool showLibrarySection = App.AppSettings.ShowLibrarySection;
-
+        private bool showLibrarySection = App.AppSettings.ShowLibrarySection;        
+        private bool rememberSidebarState = App.AppSettings.ShowLibrarySection;
```
Comment (winston-de) [inline Files/ViewModels/SettingsViewModels/PreferencesViewModel.cs:21]: Since there isn't an option anymore, could you please remove this?

**Review by winston-de — CHANGES_REQUESTED** [review]

**Comment by yaichenbaum** [comment]
@oleitao The state of sidebar items was already being saved, the issue is that collapsing and then expanding the sidebar again would reset the state.

**Pull request closed without merging**